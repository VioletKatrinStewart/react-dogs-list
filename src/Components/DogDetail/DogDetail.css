img {
  width: 500px;
  border-radius: 50%;
  border: solid black;
}

.detaildiv {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  color: white;
  font-size: large;
}
