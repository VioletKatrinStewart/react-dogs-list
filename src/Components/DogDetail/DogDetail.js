import React from 'react';
import './DogDetail.css';

export default function DogDetail({ dog }) {
  return (
    <div className="detaildiv">
      <h1> This is {dog.name} </h1>
      <img src={dog.image} />
      <p>
        {dog.name} is a {dog.breed} who is
        {dog.age} . Here is a little bit about our furry friend: ...
        {dog.bio}
      </p>
    </div>
  );
}
