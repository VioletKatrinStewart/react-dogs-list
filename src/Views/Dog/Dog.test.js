import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import DogDetail from './Dog';

it('should render an individual dog', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/dogs/1']}>
      {' '}
      <Route path="/dogs/:id">
        <DogDetail />
      </Route>
    </MemoryRouter>
  );
  await screen.findByText('This is Stinky');
  expect(container).toMatchSnapshot();
});
