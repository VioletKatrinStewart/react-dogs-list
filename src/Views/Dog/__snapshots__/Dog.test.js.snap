// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an individual dog 1`] = `
<div>
   
  <div
    class="dogdiv"
  >
    <header>
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/admin"
      >
        Admin
      </a>
    </header>
    <div
      class="detaildiv"
    >
      <h1>
         This is 
        Mittens 
         
      </h1>
      <p />
      <img
        src="https://i.insider.com/5cdeb255021b4c06ca38bbb3?width=2000&format=jpeg&auto=webp"
      />
      <p>
        Mittens 
         is a 
        Cat
         who is
         12
         . Here is a little bit about our furry friend: ...
        I am not a dog, how did I get here?
      </p>
      <button>
        Delete
      </button>
    </div>
    <a
      href="/dogs/6/edit"
    >
      Edit
    </a>
  </div>
</div>
`;
